odoo/addons/base/models/ir_binary.py

```diff
@@ -0,0 +1,162 @@
+        record.check_access_rule('read')
+
+        return record
+
+    @user_error_as_not_found
+    def _get_stream_from(
+        self, record, field='raw', filename=None, filename_field='name',
+        mimetype=None, default_mimetype='application/octet-stream',
+    ):
+        record.ensure_one()
+
+        try:
+            field_def = record._fields[field]
+        except KeyError:
+            raise UserError(f"Record has no field {field}.")
+        if field_def.type != 'binary':
+            raise UserError(f"Field '{field_def!r}' is type {field_def.type!r} but it is only possible to stream 'binary' fields.")
+        elif record._name == 'ir.attachment':
+            stream = Stream.from_attachment(record)
+        elif not field_def.attachment or field_def.compute or field_def.related:
+            stream = Stream.from_binary_field(record, field)
+        else:
+            field_attachment = self.env['ir.attachment'].sudo().search(
+                domain=[('res_model', '=', record._name),
+                        ('res_id', '=', record.id),
+                        ('res_field', '=', field)],
+                limit=1)
+            if not field_attachment.exists():
+                raise MissingError("The related attachment does not exist.")
+            stream = Stream.from_attachment(field_attachment)
+
+        if stream.type in ('data', 'path'):
+            if mimetype:
+                stream.mimetype = mimetype
+            elif not stream.mimetype:
+                if stream.type == 'data':
+                    head = stream.data[:1024]
+                else:
+                    with open(stream.path, 'rb') as file:
+                        head = file.read(1024)
+                stream.mimetype = guess_mimetype(head, default=default_mimetype)
+
+            if filename:
+                stream.download_name = filename
+            elif filename_field in record._fields:
+                stream.download_name = record[filename_field]
+            if not stream.download_name:
+                stream.download_name = f'{record._table}-{record.res_id}-{field}'
+
+            if not os.path.splitext(stream.download_name)[1]:  # missing file extension
+                stream.download_name += guess_extension(stream.mimetype) or ''
+
+        return stream
+
+    def _get_image_stream_from(
+        self, record, field='raw', filename=None, filename_field='name',
+        mimetype=None, default_mimetype='image/png', placeholder=None,
+        width=0, height=0, crop=False, quality=0,
+    ):
+        try:
+            stream = self._get_stream_from(record, field, filename, filename_field, mimetype, default_mimetype)
+        except werkzeug.exceptions.NotFound:
+            if request.params.get('download'):
+                raise
+            if not placeholder:
+                placeholder = record._get_placeholder_filename(field)
+            stream = self._get_placeholder_stream(placeholder)
+
+        if stream.type == 'url':
```

Comment:
In that case, you don't take in account  `width=0, height=0, crop=False, quality=0,` ?

Reply:
Hard to resize an external URL, we would need to request/download the resource server-side in order to resize it. Plus when you `stream.get_response()` a stream that holds an external URL all you do is `return request.redirect(url, internal=False)`

Reply:
I'll add a comment :)